add function feature to ibek support yaml

gilesknap opened the pull request.

gilesknap commented:
@coretl @niamhdougan @AlexanderWells-diamond 

So this latest allows us to define a function in the support yaml like this:
https://github.com/epics-containers/ibek-defs/blob/main/hy8401ip/hy8401ip.ibek.support.yaml#L54-L72

Use it like this:
https://github.com/epics-containers/ibek/blob/function/tests/samples/example-srrfioc08/SR-RF-IOC-08.ibek.ioc.yaml

And you get a startup script like this:
https://github.com/epics-containers/ibek/blob/function/tests/samples/example-srrfioc08/st.cmd

coretl commented:
Looks good, note that the `name` argument for Hy8401ip is not actually needed as this is not asyn, so you refer to an 8401 by card and signal rather than a name

gilesknap commented:
Updated 8401 to use the name in the database fir 8401signal

Added a naming conventions in YAML:    

-     CamelCase for names of Definitions
-     snake_case for argument names
-     lowerCamelCase for module names (this one under review)
